pkg/jobs/registry.go

```diff
@@ -179,6 +179,19 @@ func (r *Registry) MetricsStruct() *Metrics {
 	return &r.metrics
 }
 
+// CurrentlyRunningJobs returns a slice of the ids of all jobs running on this node.
+func (r *Registry) CurrentlyRunningJobs() []int64 {
+	jobs := make([]int64, len(r.mu.jobs))
```

Comment:
oh, sorry, I missed this earlier, but I think this has a race since we read the len before we acquire the lock, so if it then immediately grows, the iteration below could out-of-bounds.

We could hold the lock while allocating, or I think just switch this to `make([]int64, 0, len(...` and then use `ids = append(ids, id)` pattern below rather than baking in the exact length assumption,

Reply:
Good catch - fixed in https://github.com/cockroachdb/cockroach/pull/45094